suite-native/receive/src/components/DeviceScreenContent.tsx

```diff
@@ -89,18 +99,50 @@ export const DeviceScreenContent = ({
     isAddressRevealed,
     isPaginationEnabled,
     activePage,
+    isReceiveApproved,
+    isConfirmOnTrezorReady,
 }: DeviceScreenContentProps) => {
     const { applyStyle } = useNativeStyles();
     const deviceModel = useSelector(selectDeviceModel);
     const { fontSource, fontSize, lineWidth, lineHeight, pagerOffset } =
         deviceToContentStyles[deviceModel ?? 'T1B1'];
 
     const deviceFont = useFont(fontSource, fontSize);
+    const { translate } = useTranslate();
+    const [visibleDots, setVisibleDots] = useState<number>(1);
+    const [visualState, setVisualState] = useState<DeviceScreenVisualState>({
+        isAddressShown: false,
+        isPaginationShown: false,
+        isLoading: false,
+    });
+
+    useEffect(() => {
+        const isLoading = isAddressRevealed && !isConfirmOnTrezorReady && !isReceiveApproved;
+        const isAddressShown = isAddressRevealed && isConfirmOnTrezorReady;
```

Comment:
Please create an animation hook if possible. Similar to how it's done in bottom sheet, to separate logic of animation.

Reply:
rewritten